lndr-backend/src/Lndr/Handler/Identity.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TupleSections #-}
+
+module Lndr.Handler.Identity where
+
+import           Control.Concurrent.STM
+import           Control.Lens
+import           Control.Monad.Except
+import           Control.Monad.Reader
+import           Control.Monad.Trans.Resource
+import qualified Data.ByteString.Base64  as B64
+import qualified Data.ByteString.Lazy    as LBS
+import           Data.Pool               (withResource)
+import           Data.Text               (Text)
+import qualified Data.Text               as T
+import qualified Data.Text.Encoding      as T
+import qualified Lndr.Db                 as Db
+import           Lndr.Handler.Types
+import           Lndr.IdentityVerification
+import           Lndr.Signature
+import           Lndr.Types              as Types
+import           Lndr.Util
+import           Network.Ethereum.Web3
+import           Network.HTTP.Client
+import           Network.HTTP.Client.MultipartFormData
+import           Servant
+import           Text.EmailAddress
+import           System.Log.FastLogger
+
+verifyIdentityHandler :: IdentityVerificationRequest -> LndrHandler NoContent
+verifyIdentityHandler req@(IdentityVerificationRequest email addr info requiredDocs signature) = do
+  unless (Right addr == recoverSigner req) $ throwError (err401 {errBody = "Bad signature."})
+  (ServerState pool configTVar loggerSet) <- ask
+  config <- liftIO $ readTVarIO configTVar
+  response <- liftIO $ sendVerificationRequest config req
+
+  liftIO . mapM (\doc -> sendVerificationDocument loggerSet config (Types.id response) doc) $ idDocs info
+  
+  -- store in db
+  liftIO . withResource pool . Db.addVerificationStatus $ VerificationStatusEntry addr (Types.id response) ""
+
+  pure NoContent
+
+
+verifyIdentityCallbackHandler :: IdentityVerificationStatus -> LndrHandler NoContent
+verifyIdentityCallbackHandler status@(IdentityVerificationStatus applicantId _ _ addr _ _ _ rev) = do
+  (ServerState pool configTVar loggerSet) <- ask
+  liftIO . withResource pool . Db.addVerificationStatus $ VerificationStatusEntry addr applicantId $ reviewAnswer rev
+  pure NoContent
```

Comment:
I'm not seeing where this request is being authenticated, so it would seem like anyone could set the identity verification status for any user using this endpoint, providing the user's address, the desired status, and any `applicantId` of their choosing.

How do we verify this request is indeed coming from sumsub?

Reply:
good point. I'll have to see if I can find anything in their documentation